### DISCUSSION TITLE
The First Question about GBAnisotropy

### DISCUSSION THREAD

**Discussion by PengWei97** [idaholab/moose#19145] [Q&A Modules: Phase field] [answered]
Dear MOOSE experts,

Recently, I recently read the input file and C++ file about the anisotropy of grain boundary energy and grain boundary mobility. For [test1.i](https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/phase_field/test/tests/GBAnisotropy/test1.i) and `op_num = 3`, I found that only the three GB mobility prefactor in [anisotropy_mobility.txt](https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/phase_field/test/tests/GBAnisotropy/anisotropy_mobility.txt) are used by reading [GBAnisotropy.C](https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/phase_field/src/materials/GBAnisotropy.C) and [GBAnisotropyBase.C](https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/phase_field/src/materials/GBAnisotropyBase.C), as shown in the Fig.1.
![image](https://user-images.githubusercontent.com/56183775/137827943-5d781434-e93f-4b85-a555-6ffaa696a1b3.png)

And the output of L (the mobility of the order parameter) after the simultion of test1.i is as follow in Fig.2,
![image](https://user-images.githubusercontent.com/56183775/137827958-d49fbdc2-64f9-4afc-8026-6a079acc7cce.png)

Finially, my questions for test1.i are, 

1. The correspondence between $GBmob0_{01}, GBmob0_{02}$ , mobility prefactor of the two grain boundaries (gr0-gr1 and gr0-gr2), and the value in anisotropy_mobility.txt is shown in fig.1, but I don't know if my understanding is correct?
2. For grain gr1, the intra-grain mobility prefactor is different from the inter-grain boundary perfactor, but the intra-grain mobility prefactor is the same as the the inter-grain boundary perfactor for grain gr1. And I don't know what setting casuse this difference.

Any suggestions or recommendations to fix the problem would be greatly appreciated.

Thank you
Wei

**Comment by GiudGiud** [comment, marked as answer]
@laagesen @SudiptaBiswas @amjokisaari

**Reply by laagesen** [reply]
Wei, I ran this example and output the parameter L to the exodus file by adding `outputs = exodus` in the `CuGrGranisotropic` block. The values of L are a bit different from what is in your schematic drawing.
![Screen Shot 2021-10-25 at 5 13 14 PM](https://user-images.githubusercontent.com/3810555/138783251-770b6288-9d77-4c99-a86f-d0d41067ce3b.png)

What you see is that there is a consistent value of L being used in the gr0-gr1 interface (L is consistent between through that interface). Similarly, there is a consistent (larger) value of L being used in the gr0-gr2 interface. These different values give different interface motion for the two types of interface. (There is a third value of L being used, far away from the interfaces and shown in green, but this values doesn't affect microstructural evolution because the order parameters are 1 there).

**Reply by PengWei97** [reply]
Thank you so much, @laagesen.

> There is a third value of L being used, far away from the interfaces and shown in green, but this values doesn't affect microstructural evolution because the order parameters are 1 there

Yeah, I fully understand what you mean.

Then, what I want to ask is whether the corresponding relationship in Figure 1 is correct, especially for the parameter (1,2), 
![image](https://user-images.githubusercontent.com/56183775/139171891-452293a9-9514-4828-9e24-bc86810d237e.png)
And, if I set (1, 2) to 5.0e-6, an opposite result will appear, as shown in the figure below, which is very interesting.
![image](https://user-images.githubusercontent.com/56183775/139171984-b08521dd-3b4c-485b-8b24-4ea387482dda.png)

**Reply by laagesen** [reply]
Yes I believe what you sketched in Fig. 1 is correct. However I believe if you want to change the gr1-g2 interface mobility, you also need to change the corresponding value of the gr2-gr1 parameter. Think of this as a symmetric matrix, with symmetry about the diagonal. For the values you gave in Fig. 1, it is a symmetric matrix. However when you change gr1-gr2 without changing gr2-gr1, the matrix is no longer symmetric.

**Reply by PengWei97** [reply]
ok, thank you so much. I got it.

Now, if I want to extend this Tricrystal simulation to polycrystal simulation without using graintracker object (e.g. op_num = 5, grain_num = 5), GB mobility is a 5*5 matrix in anisotropy_mobility.txt, right? 

my results for this polycrystal simulation and GB mobility in anisotropy_mobility.txt are as follow,
![image](https://user-images.githubusercontent.com/56183775/139391184-451a3fb1-750c-47c9-b412-1981bfc8703f.png)
![image](https://user-images.githubusercontent.com/56183775/139391208-72b72186-e90c-4740-90f3-2f75c027b3f8.png)

Moreover, if I use the graintracker object to reduce the number of the order parameters in this polycrystal simultion (e.g. op_num = 4, grain_num = 5), how do I set my anisotropy_mobility.txt file to get the same result as above?

**Reply by laagesen** [reply]
You're correct about it being a 5 x 5 matrix for 5 grains, and the matrix structure you show in the figure is correct. For the 5-grain simulation, I don't think you're going to be able to reduce the number of order parameters less than 5 because the grains are not far enough apart to avoid them contacting each other. Even if you could, the additional overhead of the graintracker would probably not be worth decreased cost obtained by dropping down one order parameter. GrainTracker is meant for much larger simulations, like if you wanted to simulate 100 grains with 8-9 order parameters.

**Reply by PengWei97** [reply]
Yeah， I know it. 

I just give an example, although it is not perfect. When I use the graintracker object (e.g. 100 grains and 8 op_num), is this GB mobility matrix is 8 x 8 or 100 x 100 in anisotropy_mobility.txt ? In other words, is the information assigned to the order parameters or to each grain unique_grains in *.txt?

**Reply by laagesen** [reply]
This GB mobility matrix would be 8 x 8, the information is assigned to the order parameters rather than the unique_grains.

**Reply by PengWei97** [reply]
Ok, I already understand. Thank you very much for your careful answers.

Wei
